java/002_Add_Two_Numbers.java

```diff
@@ -0,0 +1,86 @@
+package ListNode;
+
+//2. Add Two Numbers
+/**
+ * You are given two non-empty linked lists representing two non-negative integers. The digits are stored in reverse order and each of their nodes contain a single digit. Add the two numbers and return it as a linked list.
+ * 
+ * You may assume the two numbers do not contain any leading zero, except the number 0 itself.
+ * 
+ * Input: (2 -> 4 -> 3) + (5 -> 6 -> 4)
+ * Output: 7 -> 0 -> 8
+ * 
+ * Tags:  Linked List, Math
+ * Similar Problems:  (M) Multiply Strings (E) Add Binary (E) Sum of Two Integers (E) Add Strings (M) Add Two Numbers II
+ * 
+ * @author Xinyue Zhang
+ *
+ */
+
+//ListNode class
+/**
+package ListNode;
+
+public class ListNode {
+    int val;
+    ListNode next;
+    ListNode(int x) { val = x; }
+    
+    public static ListNode createList(int val) {
+        ListNode prev = new ListNode(0);
+        ListNode node = new ListNode(0);
+        ListNode head = prev;
+        
+        node.val = val;
+        prev.next = node;
+        prev = node;
+        
+        return head.next;
+    }
+    
+    public static void insertList(int val, ListNode head) {
+        ListNode node = new ListNode(0);
+        node.val = val;
+        while (head.next != null)
+            head = head.next;
+        head.next = node;
+    }
```

Comment:
The purpose of this ListNode class is to test your code. What we want here is to give your code two initial numbers. It's not necessary to add the element to the end of the list. If you add the element to the head of list, you can manipulate one element at a time, not going through the whole list. 

Here is what you do to initialize: 
2
2 -> 4
2 -> 4 -> 3

What I said is like this:
3
4 -> 3
2 -> 4 -> 3

You can use another constructor overriding the current one:
`ListNode(int x, ListNode node) { val = x; next = node; }`